submissions/nadsatt/friends-app/modules/filter-group.js

```diff
@@ -0,0 +1,73 @@
+import * as searchFiltersModule from './search-filter-items.js';
+import * as sortFiltersModule from './sort-filter-items.js';
+import { FilterList } from './filter-list.js';
+import { IconList } from './icon-list.js';
+
+export class FilterGroup {
+    constructor(userService, pageLinkList){
+        const categories = ['sort', 'search'];
+        const sortFilterClasses = Object.values(sortFiltersModule);
+        const searchFilterClasses = Object.values(searchFiltersModule);
+
+        this.element = document.createElement('div');
+        this.element.classList.add('filter-group');
+        this.element.addEventListener('click', e => this.performFiltering(e));
+        this.element.addEventListener('input', e => this.performFiltering(e));
+
+        this.userService = userService;
+        this.pageLinkList = pageLinkList;
+
+        this.sortFilters = sortFilterClasses
+            .map(FilterClass => new FilterClass(categories[0], userService));
+        this.searchFilters = searchFilterClasses
+            .map(FilterClass => new FilterClass(categories[1], userService));
+        this.filters = this.sortFilters.concat(this.searchFilters);
+
+        this.sortFilterList = new FilterList(this.sortFilters, categories[0]);
+        this.searchFilterList = new FilterList(this.searchFilters, categories[1]);
+
+        this.sortIconList = new IconList(this.sortFilterList, categories[0]);
+        this.searchIconList = new IconList(this.searchFilterList, categories[1]);
+
+        const filterSubGroups = categories.map(category => {
+            const filterSubGroup = document.createElement('div');
+            filterSubGroup.classList.add(`${category}-filter-subgroup`);
+            filterSubGroup.innerHTML = `<h3 class="filter-subgroup__heading">${category}</h3>`;
+
+            filterSubGroup.append(this[`${category}IconList`].element,
+                                  this[`${category}FilterList`].element);
+            return filterSubGroup;
+        });
```

Comment:
Better will be if you will create separate function for this callback and then call it. More readable